spec/models/exercise_log_spec.rb

```diff
@@ -29,6 +32,78 @@
     it { should delegate_method(:name).to(:exercise).with_prefix }
   end
 
+  describe 'self.at_home' do
+    it 'returns exercise_log if it doesn\'t belong to pt_session' do
+      exercise_log = create(:exercise_log, pt_session_id: nil)
+
+      expect(ExerciseLog.at_home.first).to eq exercise_log
+    end
+
+    it 'returns empty if exercise_logs belong to pt_sessions' do
+      user = create(:user)
+      pt_session = create(:pt_session, user_id: user.id)
+      exercise_log = build(:exercise_log, pt_session_id: pt_session.id)
+
+      expect(ExerciseLog.at_home).to be_empty
+    end
+  end
+
```

Comment:
This is a great start! I'd like to see some changes. Since this method is actually a scope that returns an array of items, I'd like the tests to reflect that the expected response is an array. For example:

```
# instead of expecting a single item to be some value
expect(ExerciseLog.at_home.first) ...

# expect that the resulting array includes certain items
expect(ExerciseLog.at_home)...
```

This is more like what I'd like to see:
```
describe 'self.at_home' do
  it 'returns exercise_logs that are not associated with a pt_session' do
    exercise_log = create(:exercise_log, pt_session_id: nil)

    expect(ExerciseLog.at_home).to include(exercise_log)
  end

  it 'does not return exercise_logs that are associated with a pt_session' do
    pt_session = create(:pt_session)
    exercise_log = create(:exercise_log, pt_session_id: pt_session.id)

    expect(ExerciseLog.at_home).to_not include(exercise_log)
  end

  it 'returns an empty array if all exercise_logs belong to pt_sessions' do
    pt_session = create(:pt_session)
    exercise_log = build(:exercise_log, pt_session_id: pt_session.id)

    expect(ExerciseLog.at_home).to eq([])
  end
end
```

Reply:
sorry may I ask a question? why do we need the two later cases instead of using only one?
is `associated with a pt_session` different from `belong to pt_sessions`?

Reply:
> why do we need the two later cases instead of using only one?

When testing, I like to make sure we think through the happy path, the unhappy path, and any edge cases that may occur. Without thinking through all of these case, we get vulnerabilities and blind spots in the way our code works.  


> is associated with a pt_session different from belong to pt_sessions?

There is no technical difference. There is a conceptual semantic difference to me in that I like the `it` statement to be about the end result a user would experience. The `belongs_to` is a technical description and happens to be the way we implemented this relationship in Rails. The implementation may change some day, but the intention will not.

Reply:
thank you so much for the clear explanation.